src/main/java/lotto/service/FinalResultService.java

```diff
@@ -0,0 +1,42 @@
+package lotto.service;
+
+import lotto.domain.prize.constants.PrizeGrade;
+
+import java.util.Arrays;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Objects;
+
+public class FinalResultService {
+    private FinalResultService() {
+    }
+
+    public static EnumMap<PrizeGrade, Integer> generateResult(List<PrizeGrade> prizeGrades) {
+        EnumMap<PrizeGrade, Integer> results = new EnumMap<>(PrizeGrade.class);
+        PrizeGrade[] entirePrizeGradeArray = PrizeGrade.values();
+
+        Arrays.stream(entirePrizeGradeArray)
+                .forEach(grade -> putPrizeGradeCount(prizeGrades, results, grade));
+
+        return results;
+    }
+
+    private static void putPrizeGradeCount(
+            List<PrizeGrade> prizeGrades,
+            EnumMap<PrizeGrade, Integer> results,
+            PrizeGrade grade
+    ) {
+        results.put(grade,
+                countPrizeGrade(prizeGrades, grade));
+    }
+
+    private static Integer countPrizeGrade(
+            List<PrizeGrade> prizeGrades,
+            PrizeGrade prizeGrade
+    ) {
+        return (int) prizeGrades.stream()
+                .filter(grade -> Objects.equals(grade, prizeGrade))
+                .count();
+    }
+
```

Comment:
[필요에 따라 유용할 것 같아요!]
```suggestion
     public static EnumMap<PrizeGrade, Integer> generateResult(List<PrizeGrade> prizeGrades) {
        return EnumSet.allOf(PrizeGrade.class)
                .stream()
                .collect(toMap(Function.identity(),
                        (prizeGrade) -> countPrizeGrade(prizeGrades, prizeGrade),
                        (prev,next) -> next,
                        () -> new EnumMap<>(PrizeGrade.class));
    }

    private static Integer countPrizeGrade(
            List<PrizeGrade> prizeGrades,
            PrizeGrade prizeGrade
    ) {
        return (int) prizeGrades.stream()
                .filter(grade -> Objects.equals(grade, prizeGrade))
                .count();
    }
```
이렇게 하면 더 간단하게 표현할 수 있겠네요! 추가로 해당 결과 객체도 래핑해서 넘겨주면 더 좋을 것 같습니다 :)

Reply:
스트림의 귀재 재홍님...
allof라는 문법은 처음 접해보는 것 같아요.
아마 다음 과제에서도 EnumMap을 사용해서 여럿 핸들링 하는 요소가 많을 것으로 기대가 되는데, 이 부분 꼭 참고해서 문법적 성장을 이뤄보겠습니다!